libraries/AP_MSP/AP_MSP_Telem_DJI.cpp

```diff
@@ -128,8 +128,15 @@ MSPCommandResult AP_MSP_Telem_DJI::msp_process_out_esc_sensor_data(sbuf_t *dst)
         AP_ESC_Telem& telem = AP::esc_telem();
         int16_t highest_temperature = 0;
         telem.get_highest_motor_temperature(highest_temperature);
-        sbuf_write_u8(dst, uint8_t(highest_temperature / 100));         // deg, report max temperature
-        sbuf_write_u16(dst, uint16_t(telem.get_average_motor_rpm() * 0.1f));  // rpm, report average RPM across all motors
+
+        struct PACKED {
+            uint8_t temp;
+            uint16_t rpm;
+        } esc_sensor_data {};
+
+        esc_sensor_data.temp = uint8_t(highest_temperature * 0.01f);            // deg, report max temperature
+        esc_sensor_data.rpm = uint16_t(telem.get_average_motor_rpm() * 0.1f);   // rpm, report average RPM across all motors
+        sbuf_write_data(dst, &esc_sensor_data, sizeof(esc_sensor_data));
```

Comment:
Brace init available here:

```suggestion
        const struct PACKED {
            uint8_t temp;   // deg, report max temperature
            uint16_t rpm;   // rpm, report average RPM across all motors
        } esc_sensor_data {
            temp : uint8_t(highest_temperature * 0.01f),
            rpm   : uint16_t(telem.get_average_motor_rpm() * 0.1f);
        };

        sbuf_write_data(dst, &esc_sensor_data, sizeof(esc_sensor_data));
```